FLT/DedekindDomain/FiniteAdeleRing/BaseChange.lean

```diff
@@ -547,7 +548,65 @@ noncomputable def ProdAdicCompletions.baseChangeEquiv :
     L ⊗[K] ProdAdicCompletions A K ≃ₐ[L] ProdAdicCompletions B L :=
   AlgEquiv.ofBijective
     (SemialgHom.baseChange_of_algebraMap (ProdAdicCompletions.baseChange A K L B))
-    sorry -- #239
+    (by
+      --strategy, compose a `K`-linear iso from `L ⊗ ∏ K v` to `∏ L w`
+      -- `prod_equiv` says that `∏ L ⊗ K v ≃ ∏v∣w ∏w L w`
+      let prod_equiv := AlgEquiv.piCongrRight (fun (v: HeightOneSpectrum A)
+        ↦ adicCompletionComapAlgEquiv A K L B v)
+      -- `commute` says `L ⊗ ∏ K v ≃ ∏ L ⊗ K v`
+      let commute := tensorPi_equiv_piTensor K L (adicCompletion K (R := A))
+      -- `restrict` map restricts `prod_equiv` to `K`-linear iso
+      let restrict := (prod_equiv.restrictScalars K).toLinearEquiv
+      -- `equiv_prod` gives a map from `L ⊗ ∏ K v ≃ ∏v∣w ∏w L w`
+      let equiv_prod := commute ≪≫ₗ restrict
+      -- not picking up instance; `inst_alg` should be a `haveI`? Fails on next `haveI`
```

Comment:
`inst_alg` is data so definitely shouldn't be `have` or `haveI`. But `letI` works (and saves you from having to name the instance).